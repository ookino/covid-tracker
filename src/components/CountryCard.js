import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

const CountryCard = ({ country }) => (
  <li className="card-country__li">
    <NavLink to={`details/${country.country}`} className="card-country__a">
      <img
        src={country.countryInfo.flag}
        alt="country-flag"
        className="card-country__img"
      />
      <h1 className="card-country__h1">{country.country}</h1>
      <p className="card-country__p">{`${country.cases} cases`}</p>
    </NavLink>
  </li>
);

CountryCard.propTypes = {
  country: PropTypes.shape({
    country: PropTypes.string.isRequired,
    countryInfo: PropTypes.shape({
      flag: PropTypes.string,
    }),
    cases: PropTypes.string.isRequired,
  }).isRequired,
};

export default CountryCard;
