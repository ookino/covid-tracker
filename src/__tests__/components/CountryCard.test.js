import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import CountryCard from '../../components/CountryCard';

it('renders CountryCard Component correctly', () => {
  const country = {
    country: 'Nigeria',
    cases: 523464,
    countryInfo: {
      flag: 'flag url',
    },
  };
  const tree = renderer
    .create(
      <Router>
        <CountryCard country={country} />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
