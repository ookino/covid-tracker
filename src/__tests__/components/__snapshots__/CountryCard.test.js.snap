// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CountryCard Component correctly 1`] = `
<li
  className="card-country__li"
>
  <a
    aria-current={null}
    className="card-country__a"
    href="/details/Nigeria"
    onClick={[Function]}
  >
    <img
      alt="country-flag"
      className="card-country__img"
      src="flag url"
    />
    <h1
      className="card-country__h1"
    >
      Nigeria
    </h1>
    <p
      className="card-country__p"
    >
      523464 cases
    </p>
  </a>
</li>
`;
